workspaces/tarball/src/utils/analyzeDependencies.ts

```diff
@@ -30,33 +78,38 @@ export interface analyzeDependenciesResult {
 }
 
 export function analyzeDependencies(
-  dependencies: string[],
+  sourceDependencies: string[],
   options: analyzeDependenciesOptions
 ): analyzeDependenciesResult {
-  const { packageDeps, packageDevDeps, tryDependencies, nodeImports = {} } = options;
+  const { mama, tryDependencies } = options;
+  const { dependencies, devDependencies, nodejsImports = {} } = mama;
 
   // See: https://nodejs.org/api/packages.html#subpath-imports
-  const subpathImportsDependencies = dependencies
-    .filter((name) => isAliasDependency(name) && name in nodeImports)
-    .map((name) => buildSubpathDependency(name, nodeImports));
+  const subpathImportsDependencies = Object.fromEntries(
+    sourceDependencies
+      .filter((name) => isAliasFileModule(name) && name in nodejsImports)
+      .map((name) => buildSubpathDependency(name, nodejsImports))
+  );
   const thirdPartyDependenciesAliased = new Set(
-    subpathImportsDependencies.flat().filter((name) => !isAliasDependency(name))
+    Object.values(subpathImportsDependencies).filter((mod) => !isFile(mod))
   );
 
-  const thirdPartyDependencies = dependencies
-    .map((name) => (packageDeps.includes(name) ? name : getPackageName(name)))
-    .filter((name) => !name.startsWith("."))
-    .filter((name) => !isNodeCoreModule(name))
-    .filter((name) => !packageDevDeps.includes(name))
+  const thirdPartyDependencies = sourceDependencies
+    .map((name) => (dependencies.includes(name) ? name : getPackageName(name)))
+    .filter((name) => !isFile(name))
+    .filter((name) => !isCoreModule(name))
+    .filter((name) => !devDependencies.includes(name))
     .filter((name) => !tryDependencies.has(name));
 
```

Comment:
I think we can use a flatMap here and remove each filter so we loop only once, wdyt ?

Something like this:
```ts
const thirdPartyDependencies = sourceDependencies
  .flatMap((sourceName) => {
    const name = dependencies.includes(sourceName) ? sourceName : getPackageName(sourceName);

    if (isFile(name)) {
      return [];
    }

    if (isCoreModule(name)) {
      return [];
    }

    if (devDependencies.includes(name)) {
      return [];
    }

    if (tryDependencies.has(name)) {
      return [];
    }

    return name;
  });
```